Bundler will sometimes try and verify the dependancy tree

jcstringer opened the pull request:
This fixes a bug where gems that may not be required in an env such as development or test will be verified by bundler causing the deploy to sometimes fail.

Commit:
Bundler will sometimes try and verify the dependancy tree

This fixes a bug where gems that may not be required in an env
such as development or test will be verified by bundler causing
the deploy to sometimes fail.

jcstringer commented:
@7hunderbird this came out of a pairing session I had with @krutten tonight.

dennisjbell commented on lib/eycap/recipes/bundler.rb:
```diff
@@ -13,7 +13,7 @@
         mkdir -p #{shared_path}/bundled_gems
         if [ -f #{release_path}/Gemfile ]
-        then cd #{release_path} && bundle install --without=#{bundle_without} --binstubs #{release_path}/bin --path #{shared_path}/bundled_gems --quiet
+        then cd #{release_path} && bundle install --without=#{bundle_without} --binstubs #{release_path}/bin --path #{shared_path}/bundled_gems --quiet  --deployment
```
Line 10 above should likely have the same change applied.

Commit:
Add same logic to only_with_rvm

jcstringer commented:
@dennisjbell good catch I added that in too, let me know if I missed the gist of what you were saying but I think the 2nd change should do it. Also removed some whitespace from the original commit.

mpapis commented on lib/eycap/recipes/bundler.rb:
```diff
@@ -7,13 +7,13 @@
       only_with_rvm = <<-SHELL
         if [ -f #{release_path}/Gemfile ]
-        then cd #{release_path} && bundle install --without=#{bundle_without} --system
+        then cd #{release_path} && bundle install --without=#{bundle_without} --system --deployment
```
was this happening with rvm too? if not I'm against it as the `deployment` flag is not that perfect either (it has its downsides)

7hunderbird commented:
@mpapis out of curiosity, are you using anything to run eycap and test deploy, like Vagrant, etc?

mpapis commented:
not using eycap at all, but my experiences of mixing rvm gemsets with the `--deployment` flag are very negative, it had to be turned off explicitly in many cases - including the case when I was working with eycap

dennisjbell commented:
Chris,

In light of mpapis' comment, I reverse my previous comment for keeping the two conditional branches equivalent.  Let's just change it back to your original pull request, and we'll get that released.

Thanks,
-Dennis

Commit:
Revert "Add same logic to only_with_rvm"

This reverts commit 99bbfbdd3db35c21a7e2940a01dc43f5710983c4.

jcstringer commented:
@dennisjbell makes sense I just reverted that 2nd commit so it should be what you want now.
